From 81a48ca6a060cf006cf9eec309e726c4333a9d46 Mon Sep 17 00:00:00 2001
From: tangbin <tangbin_yewu@cmss.chinamobile.com>
Date: Tue, 6 Sep 2022 11:46:41 -0400
Subject: [PATCH] kpatch-build: Remove duplicate functions

We find that there are duplicate functions of "use_klp_arch()" &
"rhel_kernel_version_gte", and one of them are useless, so remove
them.

Signed-off-by: tangbin <tangbin_yewu@cmss.chinamobile.com>
---
 kpatch-build/kpatch-build | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/kpatch-build/kpatch-build b/kpatch-build/kpatch-build
index dc5f5a5..6a0645a 100755
--- a/kpatch-build/kpatch-build
+++ b/kpatch-build/kpatch-build
@@ -198,21 +198,6 @@ rhel_kernel_version_gte() {
         [  "${ARCHVERSION}" = "$(echo -e "${ARCHVERSION}\\n$1" | sort -rV | head -n1)" ]
 }
 
-# klp.arch relocations were supported prior to v5.8
-# and prior to 4.18.0-240.el8
-use_klp_arch()
-{
-	if kernel_is_rhel; then
-		! rhel_kernel_version_gte 4.18.0-240.el8
-	else
-		! kernel_version_gte 5.8.0
-	fi
-}
-
-rhel_kernel_version_gte() {
-        [  "${ARCHVERSION}" = "$(echo -e "${ARCHVERSION}\\n$1" | sort -rV | head -n1)" ]
-}
-
 # klp.arch relocations were supported prior to v5.8
 # and prior to 4.18.0-284.el8
 use_klp_arch()
-- 
2.18.4

